python/pyhail/dataset.py

```diff
@@ -0,0 +1,941 @@
+from pyhail.java import scala_package_object
+
+class VariantDataset:
+    def __init__(self, hc, jvds):
+        self.hc = hc
+        self.jvds = jvds
+    
+    def aggregate_intervals(self, input, condition, output):
```

Comment:
This is one model, but I'm not sure if it's pythonic.  One alternative is to expose all the functions as:
```python
def aggregate_intervals(vds, input, condition, output)
```

so you invoke:
```python
vds = hc.import_vcf("...")
vds = annotate_variants_expr(vds, "expr")
```
rather than
```python
vds = hc.import_vcf("...")
vds = vds.annotate_variants_expr("expr")
```

I think we should leave things as you have them for now, but should consider both sides of this decision before we release pyhail